digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/client/vvh/VVHClient.scala

```diff
@@ -351,33 +362,55 @@ trait VVHClientOperations {
   }
 
   protected def layerDefinition(filter: String, customFieldSelection: Option[String] = None): String = {
-    URLEncoder.encode(layerDefinitionWithoutEncoding(filter, customFieldSelection) , "UTF-8")
+    URLEncoder.encode(layerDefinitionWithoutEncoding(filter, customFieldSelection), "UTF-8")
   }
 
   protected def fetchVVHFeatures(url: String): Either[List[Map[String, Any]], VVHError] = {
     time(logger, s"Fetch VVH features with url '$url'") {
-      val fetchVVHStartTime = System.currentTimeMillis()
       val request = new HttpGet(url)
       val client = HttpClientBuilder.create().build()
-      val response = client.execute(request)
       try {
-        mapFields(parse(StreamInput(response.getEntity.getContent)).values.asInstanceOf[Map[String, Any]], url)
-      } finally {
-        response.close()
+        val response = client.execute(request)
+        if (response.getStatusLine.getStatusCode >= 300) {
```

Comment:
You should do that code inside the try finally block. To have the guarantee that the response is close is something fails on those methods.